src/Symfony/Component/Validator/Constraints/CollectionValidator.php

```diff
@@ -60,7 +60,7 @@ public function validate($value, Constraint $constraint)
                 if (count($fieldConstraint->constraints) > 0) {
                     if ($context instanceof ExecutionContextInterface) {
                         $context->getValidator()
-                            ->inContext($context)
+                            ->inContext(clone $context)
```

Comment:
This solution doesn't work. Now all constraints thrown by the validators in the `Collection` constraint will be added to the cloned context and not appear in the current context (which is the point of calling `inContext()`).